flink-runtime-web/web-dashboard/src/app/pages/job/checkpoints/detail/job-checkpoints-detail.component.ts

```diff
@@ -55,21 +59,38 @@ export class JobCheckpointsDetailComponent implements OnInit {
   }
 
   refresh() {
-    this.isLoading = true;
-    if (this.jobDetail && this.jobDetail.jid) {
-      this.jobService.loadCheckpointDetails(this.jobDetail.jid, this.checkPoint.id).subscribe(
-        data => {
-          this.checkPointDetail = data;
-          this.isLoading = false;
-          this.cdr.markForCheck();
-        },
-        () => {
-          this.isLoading = false;
-          this.cdr.markForCheck();
-        }
-      );
+      this.isLoading = true;
+      if (this.jobDetail && this.jobDetail.jid) {
+        forkJoin([
+          this.jobService.loadCheckpointConfig(this.jobDetail.jid),
+          this.jobService.loadCheckpointDetails(this.jobDetail.jid, this.checkPoint.id)
+        ]).subscribe(
+          ([config, detail]) => {
+            this.checkPointConfig = config;
+            this.checkPointDetail = detail;
+            if (this.checkPointDetail.checkpoint_type === 'CHECKPOINT') {
+              if (this.checkPointConfig.unaligned_checkpoints) {
+                this.checkPointType = 'unaligned checkpoint';
+              } else {
+                this.checkPointType = 'aligned checkpoint';
+              }
+            } else if (this.checkPointDetail.checkpoint_type === 'SYNC_SAVEPOINT') {
+              this.checkPointType = 'savepoint on cancel';
+            } else if (this.checkPointDetail.checkpoint_type === 'SAVEPOINT') {
+              this.checkPointType = 'savepoint';
+            } else {
+              this.checkPointType = '-';
+            }
+            this.isLoading = false;
+            this.cdr.markForCheck();
+          },
+          () => {
+            this.isLoading = false;
+            this.cdr.markForCheck();
+          }
+        );
+      }
     }
```

Comment:
There are two extra spaces in each line, the others look good to me

Reply:
@vthinkxie I have fixed the spaces.. my IntelliJ is not enforcing the check style, I assume it is due to the community edition doesn't support ts code auto format.

Reply:
Hi, I have a submit a commit here
https://github.com/vthinkxie/flink/commit/0a513eb9ac724b2a15cf4c95acdf22e5a9456d10
could you have a look and apply this patch, there is still some format error

Reply:
sure!